includes/admin/tools/class-llms-admin-tool-batch-eraser.php

```diff
@@ -0,0 +1,142 @@
+<?php
+/**
+ * File Summary
+ *
+ * File description.
+ *
+ * @package LifterLMS/Admin/Tools
+ *
+ * @since [version]
+ * @version [version]
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * LLMS_Admin_Tool_Batch_Eraser
+ *
+ * @since [version]
+ */
+class LLMS_Admin_Tool_Batch_Eraser extends LLMS_Abstract_Admin_Tool {
+
+	/**
+	 * Tool ID.
+	 *
+	 * @var string
+	 */
+	protected $id = 'batch-eraser';
+
+	/**
+	 * Retrieve a description of the tool
+	 *
+	 * This is displayed on the right side of the tool's list before the button.
+	 *
+	 * @since [version]
+	 *
+	 * @return string
+	 */
+	protected function get_description() {
+
+		$count = $this->get_pending_batches();
+
+		$desc  = __( 'Deletes pending batches generated by LifterLMS background processors.', 'lifterlms' );
+		$desc .= ' ';
+		// Translators: %d = the number of pending batches.
+		$desc .= sprintf(
+			_n(
+				'There is currently %d pending batch that will be deleted.',
+				'There are currently %d pending batches that will be deleted.',
+				$count,
+				'lifterlms'
+			),
+			$count
+		);
+
+		return $desc;
+
+	}
+
+	/**
+	 * Retrieve the tool's label
+	 *
+	 * The label is the tool's title. It's displayed in the left column on the tool's list.
+	 *
+	 * @since [version]
+	 *
+	 * @return string
+	 */
+	protected function get_label() {
+		return __( 'Delete processor batches', 'lifterlms' );
+	}
+
+	/**
+	 * Retrieve the tool's button text
+	 *
+	 * @since [version]
+	 *
+	 * @return string
+	 */
+	protected function get_text() {
+		return __( 'Delete batches', 'lifterlms' );
+	}
+
+	/**
+	 * Retrieve the number of pending batches.
+	 *
+	 * @since [version]
+	 *
+	 * @return int
+	 */
+	protected function get_pending_batches() {
+
+		$count = wp_cache_get( $this->id, 'llms_tool_data' );
+		if ( false === $count ) {
+
+			global $wpdb;
+			$count = absint( $wpdb->get_var( "SELECT COUNT(*) FROM {$wpdb->options} WHERE option_name LIKE '%llms_%_batch_%';" ) ); // phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery
+
+			wp_cache_set( $this->id, $count, 'llms_tool_data' );
+
+		}
+
+		return $count;
+
+	}
+
+	/**
+	 * Process the tool.
+	 *
+	 * This method should do whatever the tool actually does.
+	 *
+	 * By the time this tool is called a nonce and the user's capabilities have already been checked.
+	 *
+	 * @since [version]
+	 *
+	 * @return mixed
```

Comment:
```suggestion
	 * @return boolean
```

Reply:
Since it's an abstract I'm leaving it as "mixed" so the tool itself can do whatever it wants or needs to do. The return won't actually matter for the purposes of the abstract. I'm thinking that some tools might return a simple bool (like the one included in this PR) while others may want to return a WP_Error. 

Mostly I think the return doesn't actually matter since the return will more than likely only ever be read by tests